### DISCUSSION TITLE
Do you have challenges ideas or settings you would like to see in the generator ?

### DISCUSSION THREAD

**Discussion by maximefromont** [maximefromont/sims-build-challenge-generator#1] [Ideas]
## 👋 Welcome!
  I'm using this discussion as a place for anyone to let me know if they'd like to see more challenges or settings to be added to the generator. Feel free to :
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members and debate.

**Comment by IrisWijnands** [comment]
Some base game things that I think would be fun to add: pools, ponds, and diagonal builds! They don't require any expansions but would add a little extra challenge.

**Reply by maximefromont** [reply]
Could you describe what a diagonal build is ? Or how would it appear in the challenge generator ?

What would be the maximum ammount of pools ?
Same for ponds, what would be the maximum ammount ?
Should both of thoses items have differents sizes like houses ?

**Reply by IrisWijnands** [reply]
In the Sims 4 building happens on a grid. This means that walls and stairs are placed at straight angles, usually. However, it is also possible to place walls diagonally. These types of walls are usually used for things like bay windows and such, and not often used to build entire houses because stairs cannot be placed diagonally (so it is hard to build with them). So if you're talking about challenging, making people build entirely diagonal is the ultimate challange.

I would not put a number on pools and ponds, like with basements. I would add a binary yes/no requirement and leave it at that.
While you could specify the size, I personally don't think that would really make the challenge more fun. The fact that it has to be present at all is enough. But that is just my opinion :)

**Comment by IrisWijnands** [comment]
Some expansions and how they could add to the challenge:

Cats & Dogs = by adding cats and dogs as optional family members, it would add the need to include litter boxes, food bowls, etc.
Cottage Living = chickens, llamas, and cows are not family members so they would have to be treated differently, but by adding them it would add the need for a coop or shed. While I think it would be chaotic and fun to have to include 7 cats in a build, I think having more than one shed in there is not really going to add a lot to the experience and just take up space, since you don't do a lot of building for them.
Realm of Magic = You could add the requirement for a cauldron for fun. It does not change a lot, but it would take up space.
Laundry Day= I think adding a little 'sims need to be able to do laundry' to the challenge would be fun while also leaving space for people to make their own choices. Not only can you choose to use machines or hand washing/clothes line, you can also choose to make a dedicated laundry room or just add it to the bathroom based on how many rooms you get.

There are many other packs and ways they can add fun and variety, but these are the only ones I have so they are the only ones I can speak on.

**Comment by IrisWijnands** [comment]
While it is of course totally possible to just ignore generated rules that pertain to an expansion you don't have, I think it would feel a lot more solid if there was the possibility of choosing which expansions you do or do not want to include. And as you develop this generator further you could also make options such as diagonal building optional, for if people just don't want that.

**Reply by maximefromont** [reply]
Thank you, I am indeed planning to add an option pannel, with settings to change before launching the generation.

I will find a way to separate the normal settings and the expansions.

**Comment by IrisWijnands** [comment]
I don't know to what extent these would be fun or possible, but here are some things you could include on top of what you already have:

1) In large houses (eg. 15+ rooms) add the additional challenge of randomly assigning a number between 1-5 for example to a random room type, so one of the rules can be that you need 5 bathrooms or living rooms. I would make this an option that can be turned off just in case someone does not want that level of chaos, but I think it would be hilarious especially if you also include kitchens in this.
2) Assign a budget. I would make these somewhat dependent on house size, just to make it so you don't have only 12.000 simoleons to build a 25 room house. But then again, that could be hilarious as well.
3) Perhaps it would be fun to give people the option to toggle between the 'styles' for the house, and 'aeras'. So instead of modern country or fancy, it would be 1950s.
4) You could add a random wild card in there, just for fun. For example, an indoor garden or a home gym. Alternatively, you could make the wild card actually wild and include 'no showers, only baths' or 'no indoor cooking' etc. Just add a little bit of cursed in there for flavour.

I don't know if you should add all of these, since they could make the challenge very restrictive, but seperately I like all of them and would enjoy them in a challenge.
